ServerCore/Pages/Submissions/Index.cshtml

```diff
@@ -227,71 +530,199 @@
                                         }
                                         else if (Model.PuzzleState.SolvedTime == null)
                                         {
-                                            <text>Pending...</text>
+                                            <div class="panelitem">Pending...</div>
                                         }
                                     }
                                     else
                                     {
-                                        <text>@IndexModel.IncorrectResponseText</text>
+                                        <div class="panelitem">@IndexModel.IncorrectResponseText</div>
                                     }
                                 }
                                 else
                                 {
-                                    @ServerCore.Helpers.RawHtmlHelper.Display(response.ResponseText, Model.Event.ID, Html);
+                                    <div class="panelitem">@ServerCore.Helpers.RawHtmlHelper.Display(response.ResponseText, Model.Event.ID, Html)</div>
                                 }
                             }
-                        </td>
-                    </tr>
-                }
-            </tbody>
-        </table>
+                        </div>
+                    }
+                </div>
+            </div>
+        </div>
     </div>
-}
-
-@if (!Model.PuzzleState.IsEmailOnlyMode)
-{
-    <h3>Need some help?</h3>
-    <p>While solving this puzzle, if you've run into a bug, think there's an error, or are not having fun anymore, <a href="@mailtoUrl">use this link to contact the author for @Model.Puzzle.PlaintextName</a>!</p>
-    <p>A pre-populated email form will appear and you should add details to the message body. Remember: the author may only know specifics about this puzzle and not others in the event.</p>
-}
-
-@if (Model.Event.AllowFeedback)
-{
-    <h3>Have any feedback?</h3>
-    <p><a asp-Page="/Puzzles/SubmitFeedback" asp-route-puzzleid="@Model.Puzzle.ID">Click this link</a> to submit feedback.</p>
-}
+    <br />
 
-@section Scripts {
-    @{await Html.RenderPartialAsync("_ValidationScriptsPartial");}
-    @if (@Model.PuzzleState.LockoutExpiryTime != null)
+    @if (Model.Puzzle.CustomURL != null)
     {
-        <script type="text/javascript">
-            var unlockTime = new Date(Date.UTC(
+        <iframe id="puzzle-frame" title="@Model.Puzzle.Name" sandbox="allow-same-origin allow-scripts" src="@PuzzleHelper.GetFormattedUrl(Model.Puzzle, Model.Event.ID)"></iframe>
+        <div>Want a printable version? <a href="@PuzzleHelper.GetFormattedUrl(Model.Puzzle, Model.Event.ID)" target="blank">Click here</a> to open it!</div>
+    }
+    else if (Model.Puzzle.PuzzleFile != null)
+    {
+        <object id="pdf-frame" title="@Model.Puzzle.Name" data="@Model.Puzzle.PuzzleFile.Url.AbsoluteUri" type="application/pdf">
```

Comment:
Only do this if PuzzleFile ends with ".pdf", so that HTML files etc can be embedded.

Reply:
TODO added to code referencing #818